ml-agents.Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM nvidia/cuda:10.0-cudnn7-devel-ubuntu18.04
```

Comment:
I don't think non-NVIDIA people should install the NVIDIA CUDA-based base image. This can be separated into 3 Dockerfiles.

The first Dockerfile (ml_agents_base_nvidia.Dockerfile) should define this:
```
FROM nvidia/cuda:10.0-cudnn7-devel-ubuntu18.04
# install all CUDA related stuff...
```

The second Dockerfile (ml_agents_base_opengl.Dockerfile) should define this:
```
FROM ubuntu:18.04
# Install OpenGL packages
RUN set -x \
        && apt-get update \
        && apt-get install -y xserver-xorg-video-all libgl1-mesa-glx libgl1-mesa-dri \
        && rm -rf /var/lib/apt/lists/*
```

The third Dockerfile (ml_agents.Dockerfile) should define this:
```
FROM ml_agents_base:latest
RUN apt update && apt -y upgrade
# install relevant tools
... and so on and on...
```

The first 2 Dockerfiles should be by `docker build -t ml_agents_base:latest .`. Either the one with NVIDIA or OpenGL is build depending on the system Docker is installed on. After that the third Dockerfile should be build with `docker build -t ml_agents:latest .`. This is the final image that will be used for training.

A script (build_image.sh) can be made that does this. An example for Linux is provided below. Some changes are needed for Windows.
```
#!/bin/bash
(sudo apt-get install -y mesa-utils)
vendor=`glxinfo | grep vendor | grep OpenGL | awk '{ print $4 }'`
if [ $vendor == "NVIDIA" ]; then
    (docker build -t ml_agents_base:latest -f ml_agents_base_nvidia.Dockerfile .)
else
    (docker build -t ml_agents_base:latest -f ml_agents_base_opengl.Dockerfile .)
fi
(docker build -t ml_agents:latest -f ml_agents.Dockerfile .)
```

This way new changes to the core Dockerfile is contained in a single file. Otherwise different Dockerfiles for different environments would contain the same code. This would result in a difficult to maintain mess.

Scripts for training can be made also like this:
```
#!/bin/bash
docker run -it --rm -p 5004:5004 -p 5005:5005 -p 5006:5006 -p 6006:6006 --name ml_agents:latest -v $(pwd)/data:/unity-volume artifish $1 --docker-target-name=unity-volume --run-id=$2 --train
# Sample execution of this script: ./01_train_new_model.sh configs/RL_config.yaml rollerBall
```
or
```
FOR /F "tokens=* USEBACKQ" %%F IN (`cd`) DO (SET wd=%%F)
docker run -it --rm -p 5004:5004 -p 5005:5005 -p 5006:5006 -p 6006:6006 --name ml_agents:latest -v %wd%\data:/unity-volume ml_agents:latest %1 --docker-target-name=unity-volume --run-id=%2 --train
REM Sample execution of this script: .\01_train_new_model.cmd configs/RL_config.yaml rollerBall
```